### ISSUE TITLE
k8s 1.20.1 Cannot expose service on Azure: "failed to ensure load balancer: not a vmss instance"

### ISSUE THREAD

**Issue by dmitry-irtegov** [kubernetes/kubernetes#103348]
#### What happened:

I have an 1.20.1 cluster deployed on the Azure VMs, using in-tree Azure client.  Cluster has three node groups, a master, a Scale Set node group and Availability Set node group.

When I try to expose a service as follows:
```
kubectl apply -f https://k8s.io/examples/service/load-balancer-example.yaml
kubectl expose deployment hello-world --type=LoadBalancer --name=my-service
```
It hangs in "pending" state.  Dashboard displays the state of the service as having the error
`Error syncing load balancer: failed to ensure load balancer: not a vmss instance`

#### What you expected to happen:

The service is exposed.

#### How to reproduce it (as minimally and precisely as possible):

This does not happen on single-type worker nodes, i.e. when all nodes are Scale set or Availability Set.
It worked on similar configuration on k8s 1.19.11.

#### Anything else we need to know?:
Debug log of cluster-controller:

```
0625 08:30:28.457851       1 event.go:291] "Event occurred" object="default/my-service" kind="Service" apiVersion="v1" type="Normal" reason="EnsuringLoadBalancer" message="Ensuring load balancer"
I0625 08:30:28.504641       1 azure_backoff.go:285] LoadBalancerClient.List(dvi-6604-1624596489) success
I0625 08:30:28.504816       1 azure_loadbalancer.go:1110] reconcileLoadBalancer for service(default/my-service): lb(dvi-6604-1624596489/dvi-6604-1624596489) wantLb(true) resolved load balancer name
I0625 08:30:28.505067       1 azure_vmss.go:1382] Can not extract scale set name from ipConfigurationID (/subscriptions/b28b9faf-f59f-4bce-8609-81e4a0792cff/resourceGroups/dvi-6604-1624596489/providers/Microsoft.Network/networkInterfaces/dvi-6604-1624596489-agentNic-group1-0/ipConfigurations/ipconfig1), assuming it is managed by availability set
E0625 08:30:28.505125       1 azure_loadbalancer.go:193] reconcileLoadBalancer(default/my-service) failed: not a vmss instance
I0625 08:30:28.505181       1 controller.go:789] Finished syncing service "default/my-service" (48.00771ms)
E0625 08:30:28.505669       1 controller.go:275] error processing service default/my-service (will retry): failed to ensure load balancer: not a vmss instance
```

For comparison, debug log of 1.19.7 cluster controller on the equivalent setup:

```
I0625 09:36:50.488994       1 event.go:291] "Event occurred" object="default/my-service" kind="Service" apiVersion="v1" type="Normal" reason="EnsuringLoadBalancer" message="Ensuring load balancer"
I0625 09:36:50.489050       1 endpoints_controller.go:514] Update endpoints for default/my-service, ready: 0 not ready: 0
I0625 09:36:50.504694       1 azure_loadbalancer.go:677] reconcileLoadBalancer for service(default/my-service) - wantLb(true): started
I0625 09:36:50.514178       1 endpointslice_controller.go:297] Finished syncing service "default/my-service" endpoint slices. (25.618706ms)
I0625 09:36:50.514588       1 endpointslice_controller.go:297] Finished syncing service "default/my-service" endpoint slices. (39.9µs)
W0625 09:36:50.514596       1 endpointslice_controller.go:284] Error syncing endpoint slices for service "default/my-service", retrying. Error: EndpointSlice informer cache is out of date
I0625 09:36:50.514698       1 endpointslicemirroring_controller.go:273] syncEndpoints("default/my-service")
I0625 09:36:50.514711       1 endpointslicemirroring_controller.go:270] Finished syncing EndpointSlices for "default/my-service" Endpoints. (14.2µs)
I0625 09:36:50.515563       1 endpoints_controller.go:349] Finished syncing service "default/my-service" endpoints. (28.248008ms)
I0625 09:36:50.515668       1 endpoints_controller.go:349] Finished syncing service "default/my-service" endpoints. (76.1µs)
I0625 09:36:50.624848       1 azure_backoff.go:283] LoadBalancerClient.List(dvi-6604-19-debug-1624612154) success
I0625 09:36:50.624886       1 azure_loadbalancer.go:685] reconcileLoadBalancer for service(default/my-service): lb(dvi-6604-19-debug-1624612154/dvi-6604-19-debug-1624612154) wantLb(true) resolved load balancer name
I0625 09:36:50.705916       1 azure_wrap.go:109] Public IP "dvi-6604-19-debug-1624612154-af21cf3d613e543c4a01ab2505e71746" not found
I0625 09:36:50.705945       1 azure_loadbalancer.go:542] ensurePublicIPExists for service(default/my-service): pip(dvi-6604-19-debug-1624612154-af21cf3d613e543c4a01ab2505e71746) - creating
I0625 09:36:50.705974       1 azure_loadbalancer.go:568] service(default/my-service): pip(dvi-6604-19-debug-1624612154-af21cf3d613e543c4a01ab2505e71746) - creating as ipv4 for clusterIP:100.68.136.204
I0625 09:36:50.705981       1 azure_loadbalancer.go:571] ensurePublicIPExists for service(default/my-service): pip(dvi-6604-19-debug-1624612154-af21cf3d613e543c4a01ab2505e71746) - creating
I0625 09:36:50.866857       1 reflector.go:515] k8s.io/client-go/metadata/metadatainformer/informer.go:90: Watch close - *v1.PartialObjectMetadata total 0 items received
I0625 09:36:51.517769       1 endpointslice_controller.go:297] Finished syncing service "default/my-service" endpoint slices. (2.962601ms)
I0625 09:36:53.370243       1 azure_loadbalancer.go:1033] reconcileLoadBalancerRule lb name (dvi-6604-19-debug-1624612154) rule name (af21cf3d613e543c4a01ab2505e71746-TCP-8080)
I0625 09:36:53.370280       1 azure_loadbalancer.go:956] reconcileLoadBalancer: reconcileLoadBalancer for service(default/my-service): lb(dvi-6604-19-debug-1624612154) - updating
I0625 09:36:53.794458       1 reflector.go:515] k8s.io/client-go/informers/factory.go:134: Watch close - *v1.ReplicationController total 0 items received
I0625 09:36:53.932456       1 azure_vmss.go:931] ensuring node "dvi-6604-19-debug-1624612154-group2-0-000000" of scaleset "" in LB backendpool "/subscriptions/b28b9faf-f59f-4bce-8609-81e4a0792cff/resourceGroups/dvi-6604-19-debug-1624612154/providers/Microsoft.Network/loadBalancers/dvi-6604-19-debug-1624612154/backendAddressPools/dvi-6604-19-debug-1624612154"
I0625 09:36:54.020144       1 azure_backoff.go:88] GetVirtualMachineWithRetry(dvi-6604-19-debug-1624612154-agent-group1-0): backoff success
I0625 09:36:54.091419       1 azure_vmss.go:1052] ensureVMSSInPool: ensuring VMSS with backendPoolID /subscriptions/b28b9faf-f59f-4bce-8609-81e4a0792cff/resourceGroups/dvi-6604-19-debug-1624612154/providers/Microsoft.Network/loadBalancers/dvi-6604-19-debug-1624612154/backendAddressPools/dvi-6604-19-debug-1624612154
I0625 09:36:54.091508       1 azure_vmss.go:1070] ensureVMSSInPool: found VMAS node dvi-6604-19-debug-1624612154-agent-group1-0, will skip checking and continue
I0625 09:36:54.091576       1 azure_vmss.go:1082] ensureVMSSInPool begins to update VMSS map[dvi-6604-19-debug-1624612154-group-vmss-group2-0:true] with backendPoolID /subscriptions/b28b9faf-f59f-4bce-8609-81e4a0792cff/resourceGroups/dvi-6604-19-debug-1624612154/providers/Microsoft.Network/loadBalancers/dvi-6604-19-debug-1624612154/backendAddressPools/dvi-6604-19-debug-1624612154
I0625 09:36:54.091618       1 azure_loadbalancer.go:989] reconcileLoadBalancer for service(default/my-service): lb(dvi-6604-19-debug-1624612154) finished
I0625 09:36:54.137136       1 azure_loadbalancer.go:424] getServiceLoadBalancerStatus gets ingress IP "proper ip" from frontendIPConfiguration "af21cf3d613e543c4a01ab2505e71746" for service "default/my-service"
I0625 09:36:54.137159       1 azure_loadbalancer.go:176] EnsureLoadBalancer: reconciling security group for service "default/my-service" with IP "proper ip", wantLb = true
I0625 09:36:54.137331       1 azure_loadbalancer.go:1301] reconcileSecurityGroup for service(default/my-service): sg(dvi-6604-19-debug-1624612154-SecurityGroup) - updating
```

#### Environment:
- Kubernetes version (use `kubectl version`): 1.20.1
- Cloud provider or hardware configuration: Azure, Standard_A8_v2, one node in the scale set, one in the availability set
- OS (e.g: `cat /etc/os-release`): ubuntu 20.04 (Azure Canonical-0001-com-ubuntu-server-focal-20_04-lts  SKU)
- Kernel (e.g. `uname -a`):
- Install tools: custom
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by dmitry-irtegov** [comment]
As I understand, it was fixed by https://github.com/kubernetes/kubernetes/pull/103997

**Comment by feiskyer** [comment]
The issue has been fixed in https://github.com/kubernetes/kubernetes/pull/104068 and would be included in v1.20.10.
